pandas/core/groupby/groupby.py

```diff
@@ -867,7 +867,7 @@ def _python_agg_general(self, func, *args, **kwargs):
             try:
                 result, counts = self.grouper.agg_series(obj, f)
                 output[name] = self._try_cast(result, obj, numeric_only=True)
-            except TypeError:
+            except (IndexError, TypeError):
```

Comment:
really? how does this happen?

Reply:
Some of the test cases fail when using _python_agg_general on, I believe, an empty series. I was quite confused at first as to why my code was causing this to fail when tests were passing just fine on `var`. Then I noticed that if I removed the _cython_agg_general call (which var tries to use, and only falls back on _python_agg_general in case of an exception), additional tests would begin failing

Reply:
is this still needed? what exactly is raising IndexErrror?

Reply:
So it comes from groupby/ops.py's `_aggregate_series_pure_python` (called by `agg_series` when `_aggregate_series_fast` doesn't work). In particular if `pandas/tests/resample/test_base.py:test_resample_empty_series` is running and I don't have the cython version and I instead run the python version, the test fails due to an `IndexError` from `pandas\_libs\reduction.pyx:242` where `get_result` checks `self.ngroups > 0` and then accesses `self.bins[0]` ... I guess maybe I should update that code to check for `self.bins > 0`.

I can also just not handle `IndexError` and forget about this bug; if it comes up for std(), it would also be likely to come up for other functions in the same way... only when Cython fails and some aggregation is being done on an empty series, if I understand correctly.

Reply:
yes that is better

we need to be very clear in where exceptions are caught

Reply:
It seems that there is another issue if I merely check `self.bins.size > 0` that still causes a failure; I think it's best not to worry about this case at the time.